src/lapack/backends/cusolver/cusolver_batch.cpp

```diff
@@ -652,68 +1330,153 @@ POTRS_BATCH_LAUNCHER_USM(double, cusolverDnDpotrsBatched)
+    }
+
+UNGQR_BATCH_LAUNCHER_USM(std::complex<float>, cusolverDnCungqr)
+UNGQR_BATCH_LAUNCHER_USM(std::complex<double>, cusolverDnZungqr)
+
+#undef UNGQR_BATCH_LAUNCHER_USM
+
 // BATCH SCRATCHPAD API
 
-template <>
-std::int64_t getrf_batch_scratchpad_size<float>(sycl::queue &queue, std::int64_t m, std::int64_t n,
-                                                std::int64_t lda, std::int64_t stride_a,
-                                                std::int64_t stride_ipiv, std::int64_t batch_size) {
-    throw unimplemented("lapack", "getrf_batch_scratchpad_size");
-}
-template <>
-std::int64_t getrf_batch_scratchpad_size<double>(sycl::queue &queue, std::int64_t m, std::int64_t n,
-                                                 std::int64_t lda, std::int64_t stride_a,
-                                                 std::int64_t stride_ipiv,
-                                                 std::int64_t batch_size) {
-    throw unimplemented("lapack", "getrf_batch_scratchpad_size");
-}
-template <>
-std::int64_t getrf_batch_scratchpad_size<std::complex<float>>(sycl::queue &queue, std::int64_t m,
-                                                              std::int64_t n, std::int64_t lda,
-                                                              std::int64_t stride_a,
-                                                              std::int64_t stride_ipiv,
-                                                              std::int64_t batch_size) {
-    throw unimplemented("lapack", "getrf_batch_scratchpad_size");
-}
-template <>
-std::int64_t getrf_batch_scratchpad_size<std::complex<double>>(sycl::queue &queue, std::int64_t m,
-                                                               std::int64_t n, std::int64_t lda,
-                                                               std::int64_t stride_a,
-                                                               std::int64_t stride_ipiv,
-                                                               std::int64_t batch_size) {
-    throw unimplemented("lapack", "getrf_batch_scratchpad_size");
+template <typename Func>
+inline void getrf_batch_scratchpad_size(const char *func_name, Func func, sycl::queue &queue,
+                                        std::int64_t m, std::int64_t n, std::int64_t lda,
+                                        std::int64_t stride_a, std::int64_t stride_ipiv,
+                                        std::int64_t batch_size, int *scratch_size) {
+    queue
+        .submit([&](sycl::handler &cgh) {
+            onemkl_cusolver_host_task(cgh, queue, [=](CusolverScopedContextHandler &sc) {
+                auto handle = sc.get_handle(queue);
+                cusolverStatus_t err;
+
+                CUSOLVER_ERROR_FUNC_T(func_name, func, err, handle, m, n, nullptr, lda,
+                                      scratch_size);
+            });
+        })
+        .wait();
```

Comment:
something happened to code formatting in this function

Reply:
Fixed formatting

Reply:
Thank you for the updates.